### PR TITLE
Run CI Python tasks on Fedora container

### PR DESCRIPTION
**Pull request by bgurney-rh**
Supersedes: #30 
Related: stratis-storage/project#294

### CODE REVIEW

**Review by mulkieran — CHANGES_REQUESTED** [review]
Please grab this commit: https://github.com/stratis-storage/testing/pull/32/commits/5422e9ca3cff17e467fb249cb37c20df05d248d1. Once that settles, you'll want to add one additional change.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Ok! Now add back in the MANDATORY CHECKS USING LOWEST SUPPORTED INTERPRETER, using fedora:29 image as our best proxy for the versions we require. It won't be a perfect match, but it should be adequate.

**Comment by bgurney-rh** [comment]
The old test block:
```
          # MANDATORY CHECKS USING LOWEST SUPPORTED INTERPRETER
          - python-version: 3.6.8
            dependencies: >
              dbus-python==1.2.4
              pylint==2.4.4
              psutil==5.4.3
            task: PYTHONPATH=./src make -f Makefile lint
```

...and the new test passes, with these versions installed:
```
python3-dbus-1.2.8-3.fc29
pylint-2.1.1-2.fc29
python3-astroid-2.0.4-1.fc29
python3-psutil-5.4.3-6.fc29
```

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -19,36 +19,37 @@ jobs:
...
+              python3-isort
+            image: fedora:32
             task: make -f Makefile fmt-travis
-          # MANDATORY CHECKS USING LOWEST SUPPORTED INTERPRETER
-          - python-version: 3.6.8
-            dependencies: >
-              dbus-python==1.2.4
-              pylint==2.4.4
-              psutil==5.4.3
+          # MANDATORY CHECKS USING LOWEST DEVELOPMENT INTERPRETER
```
Comment (mulkieran) [inline .github/workflows/main.yml:33]: This should really be SUPPORTED, not DEVELOPMENT, because we never do our development this far back.

**Review by mulkieran — CHANGES_REQUESTED** [review]
The dependencies we get by using f29 are acceptable. There are actually two categories of dependencies. python3-psutil and and python3-dbus are depended on by the project's source code, these are runtime dependencies. python3-psutil is exactly right, which is good. python3-dbus is a bit higher on Fedora 29 then it should be, but that library is very stable, so likely it makes no difference. pylint is, of course, a dev dependency. So long as it can be run succesfully on the current code that's good enough.
I have a suspicion that it won't run on stratis-cli, and we'll have to do some tweaking.

The interesting question, is what version of Python is Fedora 29 running. It's not easy to tell because we're just invoking python. 

What we need is one more step, that just displays the Python version, something like:

```
- name: Display Python version
  run: python --version
```

File: .github/workflows/main.yml (hunk 1 above)
Comment (bgurney-rh) [reply .github/workflows/main.yml]: Resolved.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Best to add step to show Python version in both nightly.yml and main.yml for completeness. I think then this will be done and we can move on to the ci one. stratis-cli will be the most interesting, for sure.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Looks good! Please close and open a new PR for final review.

**Pull request closed without merging**